docs/sdk/client-side-sdks/flutter.md

```diff
@@ -0,0 +1,301 @@
+```
+
+The user object needs either a `userId`, or `isAnonymous` set to `true` for an anonymous user. 
+
+### DVC Client Builder
+
+The DVCClient can be built using the following methods:
+
+[DVC ClientBuilder class](https://github.com/DevCycleHQ/flutter-client-sdk/blob/main/lib/devcycle_flutter_client_sdk.dart#L207)
+
+| Method | Parameter | Description |
+|--------|-----------|-------------|
+| environmentKey | String | DevCycle environment key |
+| user | [DVCUser](https://github.com/DevCycleHQ/flutter-client-sdk/blob/main/lib/dvc_user.dart#L1) | DevCycle user object |
+| options | [DVCOptions](https://github.com/DevCycleHQ/flutter-client-sdk/blob/main/lib/dvc_options.dart#L1) | DevCycle options object |
+
+### DVC User Builder
+The DVC user can be built using the following methods:
+
+[DVC UserBuilder class](https://github.com/DevCycleHQ/flutter-client-sdk/blob/main/lib/dvc_user.dart#L33)
+
+| Method | Parameter | Description |
+|--------|-----------|-------------|
+| userId | String | Unique user ID |
+| isAnonymous | Bool | Boolean to indicate if the user is anonymous |
+| email | String | User's email |
+| name | String | User's name |
+| language | String | User's language |
+| country | String | User's country |
+| customData | [String: Any] | Key/value map of properties to be used for targeting |
+| privateCustomData | [String: Any] | Key/value map of properties to be used for targeting. Private properties will not be included in event logging. |
+
+### DVC Options Builder
+The SDK exposes various initialization options which can be used by passing a `DVCOptions` object to the `withOptions` method of `DVCClient.builder()`:
+
+[DVC OptionsBuilder class](https://github.com/DevCycleHQ/flutter-client-sdk/blob/main/lib/dvc_options.dart#L27)
+
+| Method | Parameter | Default | Description |
+|--------|-----------|---------|-------------|
+| flushEventsIntervalMs | Int | 10000 | Controls the interval between flushing events to the DevCycle servers in milliseconds, defaults to 10 seconds. |
+| disableEventLogging | Bool | false | Disables logging of SDK generated events (e.g. aggVariableEvaluated, aggVariableDefaulted) to DevCycle. |
+| logLevel | LogLevel | error | Set log level of the default logger. Defaults to `error`|
+| enableEdgeDB | Boolean | false | Enables the usage of EdgeDB for DevCycle that syncs User Data to DevCycle. |
+| configCacheTTL | Int | 604800000 | The maximum allowed age of a cached config in milliseconds, defaults to 7 days |
+| disableConfigCache | Bool | false | Disable the use of cached configs |
+
+### Notifying when DevCycle features are available
+
+In the initialize call there is an optional `onInitialized` parameter you can use to determine when your features have been loaded:
+
+```dart
+final _dvcClient = DVCClientBuilder()
+    .environmentKey('<DEVCYCLE_MOBILE_ENVIRONMENT_KEY>')
+    .user(DVCUserBuilder().isAnonymous(true).build())
+    .build()
+    .onInitialized((error) {
+        print(error)
+    });
```

Comment:
PR so that this code is valid 😬 https://github.com/DevCycleHQ/flutter-client-sdk/pull/51